### PR TITLE
mmseq with many subset databases

### LINKED ISSUE #238: Error in rule cluster_genes
Hello @SilasK ,

I've run into an additional error in ATLAS 2.1.4 (running on Ubuntu 18.04 on a local server, not a cluster).

The error occurs at the end of `rule cluster_genes` in the Genecatalog step:
```
rule cluster_genes:
    input: Genecatalog/all_genes/predicted_genes.faa
    output: Genecatalog/all_genes/predicted_genes.db, Genecatalog/all_genes/predicted_genes.db.dbtype, Genecatalog/all_genes/predicted_genes.db.index, Genecatalog/all_genes/predicted_genes.db.lookup, Genecatalog/all_genes/predicted_genes.db_h, Genecatalog/all_genes/predicted_genes.db_h.index, Genecatalog/clustering/protein_clusters.db, Genecatalog/clustering/protein_clusters.db.index
    log: logs/Genecatalog/clustering/cluster_proteins.log
    jobid: 8
    reason: Missing output files: Genecatalog/all_genes/predicted_genes.db.dbtype, Genecatalog/all_genes/predicted_genes.db_h, Genecatalog/all_genes/predicted_genes.db_h.index, Genecatalog/clustering/protein_clusters.db.index, Genecatalog/all_genes/predicted_genes.db.lookup, Genecatalog/all_genes/predicted_genes.db.index, Genecatalog/all_genes/predicted_genes.db, Genecatalog/clustering/protein_clusters.db; Input files updated by another job: Genecatalog/all_genes/predicted_genes.faa
    threads: 20

Activating conda environment: /Data/reference_databases/atlas/2.1.x/conda_envs/d5bf8789

# [ Everything works fine after this until: ]

Program call:
mergeclusters Genecatalog/all_genes/predicted_genes.db Genecatalog/clustering/protein_clusters.db /Analysis/jmtsuji/Documents/02_Active_workfolders/49_190205_DNAseq_ELA1718_analysis/07_atlas_2.1.4_vs1a/atlas_tmp/mmseqs/7298142864430123625/pre_clust /Analysis/jmtsuji/Documents/02_Active_workfolders/49_190205_DNAseq_ELA1718_analysis/07_atlas_2.1.4_vs1a/atlas_tmp/mmseqs/7298142864430123625/clust --threads 20 --compressed 0 -v 3 

List amount 1909375
Clustering step 1...
Clustering step 2...
Writing the results...
Time for merging files: 0h 0m 0s 612ms
Time for processing: 0h 0m 1s 831ms
Waiting at most 5 seconds for missing files.
MissingOutputException in line 62 of /Analysis/jmtsuji/miniconda3_mellea/envs/atlas_2.1.4/lib/python3.6/site-packages/atlas/rules/genecatalog.snakefile:
Missing files after 5 seconds:
Genecatalog/clustering/protein_clusters.db
This might be due to filesystem latency. If that is the case, consider to increase the wait time with --latency-wait.
Removing output files of failed job cluster_genes since they might be corrupted:
Genecatalog/all_genes/predicted_genes.db, Genecatalog/all_genes/predicted_genes.db.dbtype, Genecatalog/all_genes/predicted_genes.db.index, Genecatalog/all_genes/predicted_genes.db.lookup, Genecatalog/all_genes/predicted_genes.db_h, Genecatalog/all_genes/predicted_genes.db_h.index, Genecatalog/clustering/protein_clusters.db.index
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
```

Looking in `Genecatalog/clustering`, I see:

```
protein_clusters.db.0
protein_clusters.db.1
protein_clusters.db.10
protein_clusters.db.11
protein_clusters.db.12
protein_clusters.db.13
protein_clusters.db.14
protein_clusters.db.15
protein_clusters.db.16
protein_clusters.db.17
protein_clusters.db.18
protein_clusters.db.19
protein_clusters.db.2
protein_clusters.db.3
protein_clusters.db.4
protein_clusters.db.5
protein_clusters.db.6
protein_clusters.db.7
protein_clusters.db.8
protein_clusters.db.9
protein_clusters.db.dbtype
```

So my impression is that `rule cluster_genes` finishes properly, but maybe because of the size of the input MAG collection (~750 MAGs!), it splits the clusters DB into multiple subfiles by number. The rule expects `protein_clusters.db` but instead gets `protein_clusters.db.0` and so on.

I tried running the commands of this and the next rule (`rule get_rep_proteins`) manually on the command line. All commands finish without error. Thus, it appears that subsequent commands in mmseqs (in `rule get_rep_proteins`) are able to handle the cluster DB being split across multiple files.

Because of this, I wonder if simply changing the expected output of `rule cluster_genes` to look for files in the format of `protein_clusters.db.[#]` would fix the issue. The only potential complication with this fix is that you do not know how many `protein_clusters.db.[#]` files will be created in advance, or whether a single `protein_clusters.db` file will be created (if there are not many input MAGs).

### PR DESCRIPTION
**Pull request by SilasK**
This should Fix #238

@jmtsuji

### CODE REVIEW

**Comment by jmtsuji** [comment]
@SilasK Testing on a small metagenome now.

**Comment by jmtsuji** [comment]
@SilasK Rule `cluster_genes` now finishes successfully.

However, a similar error occurs in rule `get_rep_proteins`:
```
[Tue Oct  1 19:39:12 2019]
rule get_rep_proteins:
    input: Genecatalog/all_genes/predicted_genes, Genecatalog/clustering/mmseqs
    output: Genecatalog/orf2gene_oldnames.tsv, Genecatalog/protein_catalog.db, Genecatalog/protein_catalog.db.index, Genecatalog/representatives_of_clusters.fasta
    log: logs/Genecatalog/clustering/get_rep_proteins.log
    jobid: 55
    reason: Missing output files: Genecatalog/orf2gene_oldnames.tsv, Genecatalog/representatives_of_clusters.fasta; Input files updated by another job: Genecatalog/all_genes/predicted_genes, Genecatalog/clustering/mmseqs
    threads: 40

... [job runs] ...

MissingOutputException in line 95 of /Analysis/jmtsuji/Documents/02_Active_workfolders/58_Cfx3_genomics/Data_analysis_workflow/02_assembly_binning/atlas/atlas/rules/genecatalog.snakefile:
Missing files after 5 seconds:
Genecatalog/protein_catalog.db
```

Logfile for the job: [logs/Genecatalog/clustering/get_rep_proteins.log](https://github.com/metagenome-atlas/atlas/files/3678882/get_rep_proteins.log) -- it looks like the job finished successfully to me

Looking in `Genecatalog`:
```
$ ls Genecatalog/

all_genes              protein_catalog.db.15  protein_catalog.db.23  protein_catalog.db.31  protein_catalog.db.4   protein_catalog.db.48  protein_catalog.db.6
clustering             protein_catalog.db.16  protein_catalog.db.24  protein_catalog.db.32  protein_catalog.db.40  protein_catalog.db.49  protein_catalog.db.7
protein_catalog.db.0   protein_catalog.db.17  protein_catalog.db.25  protein_catalog.db.33  protein_catalog.db.41  protein_catalog.db.5   protein_catalog.db.8
protein_catalog.db.1   protein_catalog.db.18  protein_catalog.db.26  protein_catalog.db.34  protein_catalog.db.42  protein_catalog.db.50  protein_catalog.db.9
protein_catalog.db.10  protein_catalog.db.19  protein_catalog.db.27  protein_catalog.db.35  protein_catalog.db.43  protein_catalog.db.51  protein_catalog.db.dbtype
protein_catalog.db.11  protein_catalog.db.2   protein_catalog.db.28  protein_catalog.db.36  protein_catalog.db.44  protein_catalog.db.52
protein_catalog.db.12  protein_catalog.db.20  protein_catalog.db.29  protein_catalog.db.37  protein_catalog.db.45  protein_catalog.db.53
protein_catalog.db.13  protein_catalog.db.21  protein_catalog.db.3   protein_catalog.db.38  protein_catalog.db.46  protein_catalog.db.54
protein_catalog.db.14  protein_catalog.db.22  protein_catalog.db.30  protein_catalog.db.39  protein_catalog.db.47  protein_catalog.db.55
```

Perhaps the same strategy as used for rule `get_rep_proteins` could be used here to overcome the error?

File: atlas/rules/genecatalog.snakefile [hunk 1]
```diff
@@ -64,7 +64,7 @@ if (config['genecatalog']['clustermethod']=='linclust') or (config['genecatalog'
             faa= "Genecatalog/all_genes/predicted_genes.faa"
         output:
             db=temp(directory("Genecatalog/all_genes/predicted_genes")),
-            clusterdb = directory("Genecatalog/clustering/mmseqs"),
+            clusterdb = temp(directory("Genecatalog/clustering/mmseqs"))
```
Comment (jmtsuji) [inline atlas/rules/genecatalog.snakefile]: @SilasK This directory does not seem necessary for the user. Okay to make a temp directory and delete once ATLAS is done?

**Comment by SilasK** [comment]
Great thank you for the debugging.

**Pull request merged**